bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/XMLItem.java

```diff
@@ -571,7 +634,22 @@ public void removeChildren(String qname) {
 
         Collections.reverse(toRemove);
         for (Integer index : toRemove) {
-            children.remove(index.intValue());
+            BXML removed = children.remove(index.intValue());
+            removeParentReference(removed);
+        }
+    }
+
+    private void removeParentReference(BXML removedItem) {
+        if (removedItem.getNodeType() == ELEMENT) {
```

Comment:
Shall we return early by checking
```suggestion
        if (removedItem.getNodeType() != ELEMENT) {
```